lib/libzrepl/mgmt_conn.c

```diff
@@ -501,11 +501,29 @@ uzfs_zvol_mgmt_get_handshake_info(zvol_io_hdr_t *in_hdr, const char *name,
 	    &zinfo->checkpointed_ionum);
 	error2 = uzfs_zvol_get_last_committed_io_no(zv, DEGRADED_IO_SEQNUM,
 	    &zinfo->degraded_checkpointed_ionum);
-	if ((error1 != 0) || (error2 != 0)) {
-		LOG_ERR("Failed to read io_seqnum %s", zinfo->name);
+	if (error1 != 0) {
+		LOG_ERR("Failed to read io_seqnum %s, err1: %d err2: %d",
+		    zinfo->name, error1, error2);
 		return (-1);
 	}
 
+	/*
+	 * Success condition for error2
+	 * error2 can be 0 (or)
+	 * error2 can be ENOENT when REPLICA_VERSION <= 3
+	 */
+	if (!((error2 == 0) || (error2 == ENOENT && REPLICA_VERSION <= 3))) {
```

Comment:
REPLICA_VERSION <= 3, you should not use Macro here as it will be replaced at the compile time only. You should get the version info from the handshake message only.